Review/change performance tests so they can be used with Python 3.7.10

## Description

We need that the tests developed in the [Epic 1845](https://github.com/wazuh/wazuh-qa/issues/1845) can be run with Python 3.7.10 in order to automate their execution. This implies eliminating the use of the Walrus operator and any other tool introduced in Python after that version. 
```
E     File "/home/ec2-user/wazuh-qa/tests/performance/test_cluster/test_cluster_performance/test_cluster_performance.py", line 58
E       if (selected_conf := f"{n_workers}w_{n_agents}a") not in configurations:
E                         ^
E   SyntaxError: invalid syntax
```

The following tests and tools need to be reviewed and updated:
- [x] [Cluster performance test.](https://github.com/wazuh/wazuh-qa/tree/master/tests/performance/test_cluster)
- [x] [Cluster reliability tests.](https://github.com/wazuh/wazuh-qa/tree/master/tests/reliability/test_cluster/test_cluster_logs)
- [x] [Csv parser.](https://github.com/wazuh/wazuh-qa/blob/master/deps/wazuh_testing/wazuh_testing/tools/performance/csv_parser.py)

The tests should be executed with a Python 3.7.10 environment to ensure that everything works fine after these changes.

## Issue update

First of all, I've installed Python7.10, in order to make sure that every test can be properly executed:
```
(python3.7.10_testing) yanazaeva@pop-os:~/git/wazuh-qa/deps/wazuh_testing$ python3.7 --version
Python 3.7.10
```

### Cluster performance tests and CSV Parser
After removing the [walrus](https://realpython.com/python-walrus-operator/), the test seems to be working perfectly:
```
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.10, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/yanazaeva/git/wazuh-qa
plugins: html-3.1.1, metadata-2.0.1, testinfra-5.0.0
collected 1 item                                                                                                                                                                                                  

../../tests/performance/test_cluster/test_cluster_performance/test_cluster_performance.py Setup phase took 0:19:36s (2022/03/07 14:39:43 - 2022/03/07 14:59:19).
Stable phase took 0:10:14s (2022/03/07 14:59:19 - 2022/03/07 15:09:33).
.

========================================================================================== 1 passed, 1 warning in 0.61s ===========================================================================================
```

The report is being properly generated as well:

![image](https://user-images.githubusercontent.com/67015786/161757730-508b0fa0-8118-421d-8176-07bd75ad6873.png)

The CSV Parser module seems to be working as expected too. 

### Cluster reliability tests

After removing the walruses here, we can see that the test is working perfectly.
<details>
<summary>Click to expand first test result</summary>

```
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.10, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/yanazaeva/git/wazuh-qa
plugins: html-3.1.1, metadata-2.0.1, testinfra-5.0.0
collected 4 items                                                                                                                                                                                                 

../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_connection/test_cluster_connection.py F                                                                                                 [ 25%]
../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_error_logs/test_cluster_error_logs.py F                                                                                                 [ 50%]
../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_logs_order/test_cluster_logs_order.py F                                                                                                 [ 75%]
../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_sync/test_cluster_sync.py .                                                                                                             [100%]

==================================================================================================== FAILURES =====================================================================================================
_____________________________________________________________________________________________ test_cluster_connection _____________________________________________________________________________________________

artifacts_path = '/docs/threshold_issue/big/'

    def test_cluster_connection(artifacts_path):
        """Verify that no worker disconnects from the master once they are connected.
    
        For each worker, this test looks for the first successful connection message
        in its logs. Then it looks for any failed connection attempts after the successful
        connection found above.
    
        Args:
            artifacts_path (str): Path where folders with cluster information can be found.
        """
        if not artifacts_path:
            pytest.fail("Parameter '--artifacts_path=<path>' is required.")
    
        cluster_log_files = glob(join(artifacts_path, 'worker_*', 'logs', 'cluster.log'))
        if len(cluster_log_files) == 0:
            pytest.fail(f'No files found inside {artifacts_path}.')
    
        for log_file in cluster_log_files:
            with open(log_file) as f:
                s = mmap(f.fileno(), 0, access=ACCESS_READ)
                # Search first successful connection message.
                conn = re.search(rb'^.*Sucessfully connected to master.*$', s, flags=re.MULTILINE)
                if not conn:
                    pytest.fail(f'Could not find "Sucessfully connected to master" message in the '
                                f'{node_name.search(log_file)[1]}')
    
                # Search if there are any connection attempts after the message found above.
                if re.search(rb'^.*Could not connect to master. Trying.*$|^.*Sucessfully connected to master.*$',
                             s[conn.end():], flags=re.MULTILINE):
                    disconnected_nodes.append(node_name.search(log_file)[1])
    
        if disconnected_nodes:
>           pytest.fail(f'The following nodes disconnected from master at any point:\n- ' + '\n- '.join(disconnected_nodes))
E           Failed: The following nodes disconnected from master at any point:
E           - worker_16
E           - worker_11
E           - worker_15
E           - worker_22
E           - worker_9
E           - worker_3
E           - worker_6
E           - worker_14
E           - worker_7
E           - worker_1
E           - worker_13
E           - worker_25
E           - worker_23
E           - worker_18
E           - worker_21
E           - worker_5
E           - worker_2
E           - worker_12
E           - worker_17
E           - worker_4
E           - worker_19
E           - worker_20
E           - worker_24
E           - worker_10
E           - worker_8

../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_connection/test_cluster_connection.py:47: Failed
_____________________________________________________________________________________________ test_cluster_error_logs _____________________________________________________________________________________________

artifacts_path = '/docs/threshold_issue/big/'

    def test_cluster_error_logs(artifacts_path):
        """Look for any error messages in the logs of the cluster nodes.
    
        Any error message that is not included in the "white_list" will cause the test to fail.
        Errors found are attached to an html report if the "--html=report.html" parameter is specified.
    
        Args:
            artifacts_path (str): Path where folders with cluster information can be found.
        """
        if not artifacts_path:
            pytest.fail('Parameter "--artifacts_path=<path>" is required.')
    
        cluster_log_files = glob(join(artifacts_path, '*', 'logs', 'cluster.log'))
        if len(cluster_log_files) == 0:
            pytest.fail(f'No files found inside {artifacts_path}.')
    
        for log_file in cluster_log_files:
            with open(log_file) as f:
                s = mmap(f.fileno(), 0, access=ACCESS_READ)
                error_lines = re.findall(rb'(^.*?error.*?$)', s, flags=re.MULTILINE | re.IGNORECASE)
                if error_lines:
                    error_lines = [error for error in error_lines if not error_in_white_list(error)]
                    if error_lines:
                        nodes_with_errors.update({node_name.search(log_file)[1]: error_lines})
    
>       assert not nodes_with_errors, 'Errors were found in the "cluster.log" file of ' \
                                      'these nodes: \n- ' + '\n- '.join(nodes_with_errors)
E       AssertionError: Errors were found in the "cluster.log" file of these nodes: 
E         - worker_16
E         - worker_11
E         - worker_15
E         - worker_22
E         - worker_9
E         - worker_3
E         - worker_6
E         - worker_14
E         - worker_7
E         - worker_1
E         - worker_13
E         - worker_25
E         - worker_23
E         - worker_18
E         - worker_21
E         - worker_5
E         - worker_2
E         - worker_12
E         - worker_17
E         - worker_4
E         - worker_19
E         - worker_20
E         - worker_24
E         - worker_10
E         - worker_8
E       assert not {'worker_1': [b'2022/03/07 14:57:40 ERROR: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_1] [Main] Connecti...58:20 ERROR: [Local 596130] [Main] Error during connection with '596130': [Errno 104] Connection reset by peer."], ...}

../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_error_logs/test_cluster_error_logs.py:57: AssertionError
__________________________________________________________________________________________ test_check_logs_order_workers __________________________________________________________________________________________

artifacts_path = '/docs/threshold_issue/big/'

    def test_check_logs_order_workers(artifacts_path):
        """Check that cluster logs appear in the expected order.
    
        Check that for each group of logs (agent-info, integrity-check, etc), each message
        appears in the order it should. If any log is duplicated, skipped, etc. the test will fail.
    
        Args:
            artifacts_path (str): Path where folders with cluster information can be found.
        """
        if not artifacts_path:
            pytest.fail('Parameter "--artifacts_path=<path>" is required.')
    
        cluster_log_files = glob(os.path.join(artifacts_path, 'worker_*', 'logs', 'cluster.log'))
        if len(cluster_log_files) == 0:
            pytest.fail(f'No files found inside {artifacts_path}.')
    
        for log_file in cluster_log_files:
            failed_tasks = set()
    
            with open(log_file) as file:
                for line in file.readlines():
                    result = logs_format.search(line)
                    if result:
                        if result.group(1) in logs_order and result.group(1) not in failed_tasks:
                            tree_info = logs_order[result.group(1)]
                            for child in tree_info['tree'].children(tree_info['node']):
                                if re.search(child.tag, result.group(2)):
                                    # Current node is updated so the tree points to the next expected log.
                                    logs_order[result.group(1)]['node'] = child.identifier if \
                                        tree_info['tree'].children(child.identifier) else 'root'
                                    break
                            else:
                                # Log can be different to the expected one only if permission was not granted.
                                if "Master didn't grant permission to start a new" not in result.group(2):
                                    if node_name.search(log_file)[1] not in incorrect_order:
                                        incorrect_order[node_name.search(log_file)[1]] = []
                                    incorrect_order[node_name.search(log_file)[1]].append({
                                        'log_type': result.group(1),
                                        'found_log': result.group(0),
                                        'expected_logs': [log.tag for log in tree_info['tree'].children(tree_info['node'])]
                                    })
                                    failed_tasks.add(result.group(1))
    
            # Update status of all logs so they point to their tree root.
            for log_type, tree_info in logs_order.items():
                tree_info['node'] = 'root'
    
        if incorrect_order:
            result = ''
            for node, info in incorrect_order.items():
                result += f"\n\n{node}"
                for items in info:
                    result += '\n - Log type: {log_type}\n' \
                              '   Expected logs: {expected_logs}\n' \
                              '   Found log: {found_log}\n'.format(**items)
    
>           pytest.fail(result)
E           Failed: 
E           
E           worker_16
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:01 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_16] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_11
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:10 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_11] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_15
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:10 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_15] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_22
E            - Log type: Integrity check
E              Expected logs: ["Compressing 'files_metadata.json'.*"]
E              Found log: 2022/03/07 14:57:35 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_22] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_9
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:01 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_9] [Integrity check] Permission to synchronize granted.
E           
E            - Log type: Agent-info sync
E              Expected logs: ['Finished in .* \\(.* chunks updated\\).*']
E              Found log: 2022/03/07 14:59:02 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_9] [Agent-info sync] Permission to synchronize granted.
E           
E           
E           worker_3
E            - Log type: Integrity check
E              Expected logs: ["Compressing 'files_metadata.json'.*"]
E              Found log: 2022/03/07 14:57:35 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_3] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_6
E            - Log type: Agent-info sync
E              Expected logs: ['Finished in .* \\(.* chunks updated\\).*']
E              Found log: 2022/03/07 14:57:36 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_6] [Agent-info sync] Permission to synchronize granted.
E           
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:02 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_6] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_14
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:57:35 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_14] [Integrity check] Permission to synchronize granted.
E           
E            - Log type: Agent-info sync
E              Expected logs: ['All chunks sent.*', 'Finished in .* \\(.* chunks updated\\).*']
E              Found log: 2022/03/07 14:57:36 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_14] [Agent-info sync] Permission to synchronize granted.
E           
E           
E           worker_7
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:02 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_7] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_1
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:11 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_1] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_13
E            - Log type: Integrity check
E              Expected logs: ['Finished in .*']
E              Found log: 2022/03/07 14:57:35 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_13] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_25
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:02 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_25] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_23
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:03 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_23] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_18
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:10 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_18] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_21
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:10 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_21] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_5
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:01 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_5] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_2
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:01 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_2] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_12
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:57:35 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_12] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_17
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:10 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_17] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_4
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:10 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_4] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_19
E            - Log type: Integrity check
E              Expected logs: ['Finished in .*']
E              Found log: 2022/03/07 14:57:35 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_19] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_20
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:57:35 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_20] [Integrity check] Permission to synchronize granted.
E           
E            - Log type: Agent-info sync
E              Expected logs: ['Finished in .* \\(.* chunks updated\\).*']
E              Found log: 2022/03/07 14:59:11 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_20] [Agent-info sync] Permission to synchronize granted.
E           
E           
E           worker_24
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:10 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_24] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_10
E            - Log type: Integrity check
E              Expected logs: ['Zip file sent to master.']
E              Found log: 2022/03/07 14:59:10 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_10] [Integrity check] Permission to synchronize granted.
E           
E           
E           worker_8
E            - Log type: Integrity check
E              Expected logs: ['Finished in .*']
E              Found log: 2022/03/07 14:59:01 DEBUG: [Worker CLUSTER-Workload_benchmarks_metrics_B100_manager_8] [Integrity check] Permission to synchronize granted.

../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_logs_order/test_cluster_logs_order.py:99: Failed
============================================================================================= short test summary info =============================================================================================
FAILED ../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_connection/test_cluster_connection.py::test_cluster_connection - Failed: The following nodes disconnected from master at any point:
FAILED ../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_error_logs/test_cluster_error_logs.py::test_cluster_error_logs - AssertionError: Errors were found in the "cluster.log" file of these...
FAILED ../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_logs_order/test_cluster_logs_order.py::test_check_logs_order_workers - Failed: 
========================================================================================== 3 failed, 1 passed in 28.82s ===========================================================================================
```

</details>

<details>
<summary>Click to expand the second test result</summary>

```
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.10, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/yanazaeva/git/wazuh-qa
plugins: html-3.1.1, metadata-2.0.1, testinfra-5.0.0
collected 4 items                                                                                                                                                                                                 

../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_connection/test_cluster_connection.py .                                                                                                 [ 25%]
../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_error_logs/test_cluster_error_logs.py .                                                                                                 [ 50%]
../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_logs_order/test_cluster_logs_order.py F                                                                                                 [ 75%]
../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_sync/test_cluster_sync.py .                                                                                                             [100%]

==================================================================================================== FAILURES =====================================================================================================
__________________________________________________________________________________________ test_check_logs_order_workers __________________________________________________________________________________________

artifacts_path = '/docs/final2_agent_groups/artifacts'

    def test_check_logs_order_workers(artifacts_path):
        """Check that cluster logs appear in the expected order.
    
        Check that for each group of logs (agent-info, integrity-check, etc), each message
        appears in the order it should. If any log is duplicated, skipped, etc. the test will fail.
    
        Args:
            artifacts_path (str): Path where folders with cluster information can be found.
        """
        if not artifacts_path:
            pytest.fail('Parameter "--artifacts_path=<path>" is required.')
    
        cluster_log_files = glob(os.path.join(artifacts_path, 'worker_*', 'logs', 'cluster.log'))
        if len(cluster_log_files) == 0:
            pytest.fail(f'No files found inside {artifacts_path}.')
    
        for log_file in cluster_log_files:
            failed_tasks = set()
    
            with open(log_file) as file:
                for line in file.readlines():
                    result = logs_format.search(line)
                    if result:
                        if result.group(1) in logs_order and result.group(1) not in failed_tasks:
                            tree_info = logs_order[result.group(1)]
                            for child in tree_info['tree'].children(tree_info['node']):
                                if re.search(child.tag, result.group(2)):
                                    # Current node is updated so the tree points to the next expected log.
                                    logs_order[result.group(1)]['node'] = child.identifier if \
                                        tree_info['tree'].children(child.identifier) else 'root'
                                    break
                            else:
                                # Log can be different to the expected one only if permission was not granted.
                                if "Master didn't grant permission to start a new" not in result.group(2):
                                    if node_name.search(log_file)[1] not in incorrect_order:
                                        incorrect_order[node_name.search(log_file)[1]] = []
                                    incorrect_order[node_name.search(log_file)[1]].append({
                                        'log_type': result.group(1),
                                        'found_log': result.group(0),
                                        'expected_logs': [log.tag for log in tree_info['tree'].children(tree_info['node'])]
                                    })
                                    failed_tasks.add(result.group(1))
    
            # Update status of all logs so they point to their tree root.
            for log_type, tree_info in logs_order.items():
                tree_info['node'] = 'root'
    
        if incorrect_order:
            result = ''
            for node, info in incorrect_order.items():
                result += f"\n\n{node}"
                for items in info:
                    result += '\n - Log type: {log_type}\n' \
                              '   Expected logs: {expected_logs}\n' \
                              '   Found log: {found_log}\n'.format(**items)
    
>           pytest.fail(result)
E           Failed: 
E           
E           worker_1
E            - Log type: Agent-info sync
E              Expected logs: ['Finished in .* \\(.* chunks sent\\).*']
E              Found log: 2022/03/29 08:36:53 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B122_manager_1] [Agent-info sync] Finished in 0.002s. Updated 0 chunks.
E           
E            - Log type: Integrity sync
E              Expected logs: ['Files to create: [0-9]* \\| Files to update: [0-9]* \\| Files to delete: [0-9]* \\| Files to send: 0.*', 'Files to create: 0 \\| Files to update: 0 \\| Files to delete: 0 \\| Files to send: [1-9][0-9]*.*', 'Files to create: [0-9]* \\| Files to update: [0-9]* \\| Files to delete: [0-9]* \\| Files to send: [1-9][0-9]*.*']
E              Found log: 2022/03/29 08:42:16 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B122_manager_1] [Integrity sync] Files to create: 0 | Files to update: 1 | Files to delete: 0
E           
E           
E           worker_2
E            - Log type: Agent-info sync
E              Expected logs: ['Finished in .* \\(.* chunks sent\\).*']
E              Found log: 2022/03/29 08:36:52 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B122_manager_2] [Agent-info sync] Finished in 0.002s. Updated 0 chunks.
E           
E            - Log type: Integrity sync
E              Expected logs: ['Files to create: [0-9]* \\| Files to update: [0-9]* \\| Files to delete: [0-9]* \\| Files to send: 0.*', 'Files to create: 0 \\| Files to update: 0 \\| Files to delete: 0 \\| Files to send: [1-9][0-9]*.*', 'Files to create: [0-9]* \\| Files to update: [0-9]* \\| Files to delete: [0-9]* \\| Files to send: [1-9][0-9]*.*']
E              Found log: 2022/03/29 08:42:15 INFO: [Worker CLUSTER-Workload_benchmarks_metrics_B122_manager_2] [Integrity sync] Files to create: 0 | Files to update: 1 | Files to delete: 0

../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_logs_order/test_cluster_logs_order.py:99: Failed
============================================================================================= short test summary info =============================================================================================
FAILED ../../tests/reliability/test_cluster/test_cluster_logs/test_cluster_logs_order/test_cluster_logs_order.py::test_check_logs_order_workers - Failed: 
=========================================================================================== 1 failed, 3 passed in 0.31s ===========================================================================================
```

</details>

The HTML report is properly created as well:

![image](https://user-images.githubusercontent.com/67015786/161760015-85db03a4-f18f-4dff-aa3c-0a1b97eb0d2b.png)

--- a/tests/performance/test_cluster/test_cluster_performance/test_cluster_performance.py
+++ b/tests/performance/test_cluster/test_cluster_performance/test_cluster_performance.py
@@ -55,7 +55,8 @@ def test_cluster_performance(artifacts_path, n_workers, n_agents):
         pytest.fail("Parameters '--artifacts_path=<path> --n_workers=<n_workers> --n_agents=<n_agents>' are required.")
 
     # Check if there are threshold data for the specified number of workers and agents.
-    if (selected_conf := f"{n_workers}w_{n_agents}a") not in configurations:
+    selected_conf = f"{n_workers}w_{n_agents}a"
+    if selected_conf not in configurations:
         pytest.fail(f"This is not a supported configuration: {selected_conf}. "
                     f"Supported configurations are: {', '.join(configurations.keys())}.")
 
@@ -97,6 +98,7 @@ def test_cluster_performance(artifacts_path, n_workers, n_agents):
     finally:
         # Add useful information to report as stdout
         try:
+            print('\n')
             print(f'Setup phase took {get_datetime_diff(cluster_info["phases"]["setup_phase"])}s '
                   f'({cluster_info["phases"]["setup_phase"][0]} - {cluster_info["phases"]["setup_phase"][1]}).')
             print(f'Stable phase took {get_datetime_diff(cluster_info["phases"]["stable_phase"])}s '

--- a/tests/reliability/test_cluster/test_cluster_logs/test_cluster_connection/test_cluster_connection.py
+++ b/tests/reliability/test_cluster/test_cluster_logs/test_cluster_connection/test_cluster_connection.py
@@ -25,14 +25,16 @@ def test_cluster_connection(artifacts_path):
     if not artifacts_path:
         pytest.fail("Parameter '--artifacts_path=<path>' is required.")
 
-    if len(cluster_log_files := glob(join(artifacts_path, 'worker_*', 'logs', 'cluster.log'))) == 0:
+    cluster_log_files = glob(join(artifacts_path, 'worker_*', 'logs', 'cluster.log'))
+    if len(cluster_log_files) == 0:
         pytest.fail(f'No files found inside {artifacts_path}.')
 
     for log_file in cluster_log_files:
         with open(log_file) as f:
             s = mmap(f.fileno(), 0, access=ACCESS_READ)
             # Search first successful connection message.
-            if not (conn := re.search(rb'^.*Sucessfully connected to master.*$', s, flags=re.MULTILINE)):
+            conn = re.search(rb'^.*Sucessfully connected to master.*$', s, flags=re.MULTILINE)
+            if not conn:
                 pytest.fail(f'Could not find "Sucessfully connected to master" message in the '
                             f'{node_name.search(log_file)[1]}')
 

--- a/tests/reliability/test_cluster/test_cluster_logs/test_cluster_error_logs/test_cluster_error_logs.py
+++ b/tests/reliability/test_cluster/test_cluster_logs/test_cluster_error_logs/test_cluster_error_logs.py
@@ -41,13 +41,15 @@ def test_cluster_error_logs(artifacts_path):
     if not artifacts_path:
         pytest.fail('Parameter "--artifacts_path=<path>" is required.')
 
-    if len(cluster_log_files := glob(join(artifacts_path, '*', 'logs', 'cluster.log'))) == 0:
+    cluster_log_files = glob(join(artifacts_path, '*', 'logs', 'cluster.log'))
+    if len(cluster_log_files) == 0:
         pytest.fail(f'No files found inside {artifacts_path}.')
 
     for log_file in cluster_log_files:
         with open(log_file) as f:
             s = mmap(f.fileno(), 0, access=ACCESS_READ)
-            if error_lines := re.findall(rb'(^.*?error.*?$)', s, flags=re.MULTILINE | re.IGNORECASE):
+            error_lines = re.findall(rb'(^.*?error.*?$)', s, flags=re.MULTILINE | re.IGNORECASE)
+            if error_lines:
                 error_lines = [error for error in error_lines if not error_in_white_list(error)]
                 if error_lines:
                     nodes_with_errors.update({node_name.search(log_file)[1]: error_lines})

--- a/tests/reliability/test_cluster/test_cluster_logs/test_cluster_logs_order/test_cluster_logs_order.py
+++ b/tests/reliability/test_cluster/test_cluster_logs/test_cluster_logs_order/test_cluster_logs_order.py
@@ -52,7 +52,8 @@ def test_check_logs_order_workers(artifacts_path):
     if not artifacts_path:
         pytest.fail('Parameter "--artifacts_path=<path>" is required.')
 
-    if len(cluster_log_files := glob(os.path.join(artifacts_path, 'worker_*', 'logs', 'cluster.log'))) == 0:
+    cluster_log_files = glob(os.path.join(artifacts_path, 'worker_*', 'logs', 'cluster.log'))
+    if len(cluster_log_files) == 0:
         pytest.fail(f'No files found inside {artifacts_path}.')
 
     for log_file in cluster_log_files:
@@ -60,7 +61,8 @@ def test_check_logs_order_workers(artifacts_path):
 
         with open(log_file) as file:
             for line in file.readlines():
-                if result := logs_format.search(line):
+                result = logs_format.search(line)
+                if result:
                     if result.group(1) in logs_order and result.group(1) not in failed_tasks:
                         tree_info = logs_order[result.group(1)]
                         for child in tree_info['tree'].children(tree_info['node']):

--- a/tests/reliability/test_cluster/test_cluster_logs/test_cluster_sync/test_cluster_sync.py
+++ b/tests/reliability/test_cluster/test_cluster_logs/test_cluster_sync/test_cluster_sync.py
@@ -32,14 +32,16 @@ def test_cluster_sync(artifacts_path):
     if not artifacts_path:
         pytest.fail('Parameter "--artifacts_path=<path>" is required.')
 
-    if len(cluster_log_files := glob(join(artifacts_path, 'worker_*', 'logs', 'cluster.log'))) == 0:
+    cluster_log_files = glob(join(artifacts_path, 'worker_*', 'logs', 'cluster.log'))
+    if len(cluster_log_files) == 0:
         pytest.fail(f'No files found inside {artifacts_path}.')
 
     repeat_counter = 0
     for log_file in cluster_log_files:
         with open(log_file) as f:
             s = mmap(f.fileno(), 0, access=ACCESS_READ)
-            if not (sync_logs := integrity_regex.findall(s)):
+            sync_logs = integrity_regex.findall(s)
+            if not sync_logs:
                 pytest.fail(f'No integrity sync logs found in {node_name.search(log_file)[1]}')
 
             for i in range(len(sync_logs)):

